examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+"""
+
+normalized_all_data = tp.glue(normalized_features, all_data["TX_FRAUD"])
+
+"""
+## Split dataset into a train, validation and test set
+
+To evaluate the quality of our machine learning model, we need training, validation and test sets. Since the system is dynamic (new fraud patterns are being created all the time), it is important for the training set to come before the validation set, and the validation set come before the testing set:
+
+- **Training:** April 8, 2018 to July 31, 2018
+- **Validation:** August 1, 2018 to August 31, 2018
+- **Testing:** September 1, 2018 to September 30, 2018
+
+For the example to run faster, we will effectively reduce the size of the training set to:
+- **Training:** July 1, 2018 to July 31, 2018
+"""
+
+# begin_train = datetime.datetime(2018, 4, 8).timestamp() # Full training dataset
+begin_train = datetime.datetime(
+    2018, 7, 1
+).timestamp()  # Reduced training dataset
+begin_valid = datetime.datetime(2018, 8, 1).timestamp()
+begin_test = datetime.datetime(2018, 9, 1).timestamp()
+
+is_train = (normalized_all_data.timestamps() >= begin_train) & (
+    normalized_all_data.timestamps() < begin_valid
+)
+is_valid = (normalized_all_data.timestamps() >= begin_valid) & (
+    normalized_all_data.timestamps() < begin_test
+)
+is_test = normalized_all_data.timestamps() >= begin_test
+
+"""
+`is_train`, `is_valid` and `is_test` are boolean features overtime that indicate the limit of the tree folds. Let's plot them.
+"""
+
+tp.plot(
+    [
+        is_train.rename("is_train"),
+        is_valid.rename("is_valid"),
+        is_test.rename("is_test"),
+    ]
+)
+
+"""
+We filter the input features and label in each fold.
+"""
+
+train_ds_eventset = normalized_all_data.filter(is_train)
+valid_ds_eventset = normalized_all_data.filter(is_valid)
+test_ds_eventset = normalized_all_data.filter(is_test)
+
+print(f"Training examples: {train_ds_eventset.num_events()}")
+print(f"Validation examples: {valid_ds_eventset.num_events()}")
+print(f"Testing examples: {test_ds_eventset.num_events()}")
+
+"""
+It is important to split the dataset **after** the features have been computed because some of the features for the training dataset are computed from transactions during the training window.
+
+## Create TensorFlow datasets
+
+We convert the datasets from event-sets to TensorFlow Datasets as Keras consumes them natively.
+"""
+
+non_batched_train_ds = tp.to_tensorflow_dataset(train_ds_eventset)
+non_batched_valid_ds = tp.to_tensorflow_dataset(valid_ds_eventset)
+non_batched_test_ds = tp.to_tensorflow_dataset(test_ds_eventset)
+
+"""
+The following processing steps are applied using TensorFlow dataset:
+
+1. The features and labels are separated using `extract_features_and_label` in the format that Keras expects.
+1. The dataset is batched, which means that the examples are grouped into mini-batches.
+1. The training examples are shuffled to improve the quality of mini-batch training.
```

Comment:
fix numbering

Reply:
This is a markdown content. Afak, this .py is automatically converted into a notebook. Triple quote comments are MD boxes.